sessionmanager.md

```diff
@@ -0,0 +1,117 @@
+# `SessionManager` for @azure/service-bus v7
+
+## Introduction
+
+Sessions allow a user to logically subdivide a queue or subscription
+into multiple smaller queues. This is useful for customers that would 
+like to allow for many smaller streams of messages within a single 
+queue or subscription.
+
+## Use case
+
+A simple use case would be having a set of documents. Users can "edit"
+documents, and each edit is stored as a message within Service Bus.
+
+```javascript
+{ 
+  sessionId: "the document name"
+  body: {
+    mode: "edit",
+    content: "content of the edit"
+  }  
+}
+```
+
+Multiple edits or transformations could be sent for a document, at any time.
+
+To handle this the user would need to write code similar to this:
+
+```javascript
+
+  while (true) {    
+    // some sort of check to make sure we're not processing
+    // too many sessions at a time. The odds of having more documents
+    // than available processors is _high_.
+    acquireSemaphore();
+
+    // by not specifying a session id we will get the next unlocked session from service bus.
+    const sessionReceiver: SessionReceiver = serviceBusClient.getSessionReceiver("queue", "peekLock");
+    startSessionLockRenewal(sessionReceiver);
+
+    sessionReceiver.subscribe(messageHandlers);
+
+    // ...after some predefined period of time
+    sessionReceiver.close();
+
+    stopSessionLockRenewal(sessionReceiver);
+    releaseSemaphore();
+  }
+```
+
+## Proposal
+
+In the same vein as what we provide in `Receiver.subscribe()` we can provide a 
+`SessionManager` that would support the use case where the user would like 
+to read messages, in parallel, from a number of sessions while controlling
+the amount of parallelism.
+
+This would simplify session usage to the point where it becomes more accessible, 
+allowing more customers to use this powerful feature.
+
+The proposed interface for this change:
+
+```javascript
+class ServiceBusClient {
+  // a session manager is conceptually a receiver, but does not have to 
+  // obey the same interface.
+  createSessionManager("queue", "peekLock", sessionManagerOptions): SessionManager;
+  createSessionManager("topic", "subscription", "peekLock", sessionManagerOptions): SessionManager;
```

Comment:
In the current proposal, is there a way where user can pass the sessionId, if they set `concurrentReceivers=1`?

Reply:
In that case they're better served just using a plain SessionReceiver (ie, getSessionReceiver( { sessionId: "already known session id" });

Reply:
This is a difference between .NET and JS - the user doesn't need to use the SessionManager to get a push model. They can just use any Receiver type (including SessionReceiver).

Reply:
@ShivangiReja  - incidentally - does the fact that you have to do this `concurrentReceivers` thing make you want to change anything? It does seem kind of clunky, but that might be because we have a different design in JS so my thoughts are biased.

Reply:
(resolving this since my question is not really a blocker for this proposal)